ui/pager.go

```diff
@@ -451,3 +464,54 @@ func glamourRender(m pagerModel, markdown string) (string, error) {
 
 	return content.String(), nil
 }
+
+func (m *pagerModel) initWatcher() {
+	var err error
+	m.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		log.Error("error creating fsnotify watcher", "error", err)
+	}
+}
+
+func (m *pagerModel) watchFile() tea.Msg {
+	path := m.relFilePath()
+
+	if err := m.watcher.Add(path); err != nil {
+		log.Error("error adding file to fsnotify watcher", "error", err)
+		return nil
+	}
+
+	log.Info("fsnotify watching file", "file", path)
+
+	for {
+		select {
+		case event, ok := <-m.watcher.Events:
+			if !ok || !event.Has(fsnotify.Write) {
+				continue
+			}
+			log.Debug("fsnotify event", "file", path, "event", event.Op)
+			return reloadKeyMsg
+		case err, ok := <-m.watcher.Errors:
+			if !ok {
+				continue
+			}
+			log.Debug("fsnotify error", "file", path, "error", err)
+		}
+	}
```

Comment:
The usual way to watch for these channels would be inside a goroutine. It wasn't clear for me how to address that though, as I don't know if there is any way to send a message from a remote context (another goroutine), specially considering this channel could receive multiple messages.

Given Bubbletea _appears_ to require an inline `return` here to force an update, I had to make this loop inline inside a `cmd`. Let me know if we have a better pattern for this, or if you have a better idea on how to implement it.

Reply:
I _think_ this is fine, Bubble Tea commands run in goroutines